File: LeetCode/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        n = len(nums)
        mp = Counter(nums)
        
        if len(mp) == n:
            return 0
        
        ans = 0
        while nums:
            for i in range(3):
                if nums:
                    first = nums.pop(0)
                    mp[first] -= 1
                    if mp[first] == 0:
                        del mp[first]
            ans += 1
            
            if len(mp) == len(nums):
                break
        
        return ans

File: LeetCode/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> mp;
        for (int x : nums) mp[x]++;
        if (mp.size() == n) return 0;
        int ans = 0;
        while (!nums.empty()) {
            for (int i = 0; i < 3 && !nums.empty(); i++) {
                mp[nums[0]]--;
                if (mp[nums[0]] == 0) {
                    mp.erase(nums[0]);
                }
                nums.erase(nums.begin());
            }
            ans++;
            if (mp.size() == nums.size()) break;
        }
        return ans;
    }
};